### DISCUSSION TITLE
Integrated jobs (git workflows) also run on forks.

### DISCUSSION THREAD

**Discussion by return42** [searx/searx#2551] [Development]
For me it is not clear, is this a side effect or is there any use of?

By example: I have a fork and my master branch is a brand [see last commit on my master](https://github.com/return42/searx/commits/master).

Every time I rebase my master on searx/master the git workflow starts to run the tests and builds .. since it is a brand this job will fail and I got a mail with the subject:

    [return42/searx] Run failed: Integration - gh-pages (nnnnnnn)

It is not the mail that bothers me, it is the thought that the builds (in forks) might be a huge waste of energy without any ecological sense.

Thoughts?

**Comment by dalf** [comment]
> it is the thought that the builds (in forks) might be a huge waste of energy without any ecological sense.

* Various optimizations have been applied to reduce the run time, for example the cache (even if right now, there is an issue).
* If your fork is not a perfect clone of searx/searx, it makes sense to check the tests, update the documentation.
* You can disable the workflow in your fork: https://docs.github.com/en/actions/managing-workflow-runs/disabling-and-enabling-a-workflow#disabling-a-workflow

**Comment by return42** [comment]
@unixfox do you like to move this issue to the discussion section for development / thanks!

**Reply by dalf** [reply]
I guess https://docs.github.com/en/actions/managing-workflow-runs/disabling-and-enabling-a-workflow#disabling-a-workflow doesn't answer your request ?

**Reply by return42** [reply]
> doesn't answer your request ?

It does, everything is fine ..I thought it should not be classified as issue since it was a missing development knowledge of mine.
